framework/packages/abstract-core/src/objects/module.rs

```diff
@@ -477,17 +477,15 @@ impl Default for Monetization {
 pub type ModuleMetadata = String;
 
 /// Generate salt helper
-pub fn generate_module_salt(block_height: u64, account_id: &AccountId) -> Binary {
-    let mut salt = [0; 40];
+pub fn generate_module_salt(account_id: &AccountId, nounce: u8) -> Binary {
+    let mut salt = [0; 33];
     // 0..32 bytes for account_id
     let account_id = sha256::digest(account_id.to_string());
     let accound_id_bytes: &mut [u8] = &mut salt[0..32];
     accound_id_bytes.copy_from_slice(&account_id.as_bytes()[0..32]);
 
-    // 0..8 bytes for block height
-    let block_height_bytes = &mut salt[32..40];
-    block_height_bytes.copy_from_slice(&block_height.to_be_bytes());
-
+    // 1 bytes for nounce
+    salt[32] = nounce;
```

Comment:
This is kind of unpredictable as it messes with the generated hash. Maybe we can concat the u8 to the account_id string and hash that?

Reply:
I don't get how it makes it unpredictable?

Reply:
It's always 32 bytes for account id and 1 byte for nonce, which is equal to the amount you reinstalled the app